### PR TITLE
Dfx 15

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We would like to keep `dfx` up to date.

# Changes
- Update `dfx` from version `0.14.4` to `0.15.1`
- Install `dfx nns` as an extension, as it is no longer in `dfx` core.

# Tests
- See CI
- [Works with the nns-dapp.](https://github.com/dfinity/nns-dapp/pull/3716)

### CODE REVIEW

File: bin/dfx-ii-create-dummy-anchor [hunk 1]
```diff
@@ -31,12 +31,24 @@ trap clean_up EXIT
 dfx identity export "$DFX_IDENTITY" | openssl ec -out "$IDENTITY_PUB_KEY_FILE" -pubout -outform der 2>/dev/null
```
Comment (bitdivine) [inline bin/dfx-ii-create-dummy-anchor:32]: Funny thing:  The path to the candid is specified in dfx.json, yet not picked up.  If I specify the path explicitly, fields are decoded correctly.

File: bin/dfx-network-deploy [hunk 2]
```diff
@@ -53,7 +53,7 @@ if [[ "$DFX_NETWORK" == "local" ]]; then
   dfx-canister-set-id --canister_name nns-dapp --canister_id qsgjb-riaaa-aaaaa-aaaga-cai
   dfx-canister-set-id --canister_name internet_identity --canister_id qhbym-qaaaa-aaaaa-aaafq-cai
 
-  dfx-software-internet-identity-install --release "${DFX_II_RELEASE:-pinned}" --flavor "dev" --network local
+  dfx-software-internet-identity-install --release "${DFX_II_RELEASE:-pinned}" --flavor "dev" --network local --mode reinstall
```
Comment (bitdivine) [inline bin/dfx-network-deploy:56]: Might not be needed.  There is a possibility that upgrading will cause stable memory corruption.

File: bin/dfx-nns-import [hunk 3]
```diff
@@ -45,9 +45,15 @@ export DFX_NETWORK
 # works as long as the .did files are not moved in the ic repo.
 DFX_IC_COMMIT="$DFX_IC_COMMIT" dfx nns import --network-mapping "$DFX_NETWORK=mainnet"
```
Comment (bitdivine) [inline bin/dfx-nns-import:47]: As in the comment, `dfx nns import` no longer provides the II candid.

**Review by lmuntaner — APPROVED** [review]
LGTM!

**Pull request merged**